### PR TITLE
#44 Travis CI integration & Jest for Unit Testing

### PR DESCRIPTION
**Pull request by JeremiahTee**
-added jest dependency for back-end, for unit testing purposes
     - so far it only has spec.js which will return True if jest correctly works
- integrated Travis CI by including a .travis.yml file
    - this yml file cd's into Server (back-end), runs npm install & test, then cd to Client (front-end) and runs npm install
    - finally, it spins up the docker containers by running docker-compose up and gracefully shuts down with docker-compose down

This resolves Issue #44

### CODE REVIEW

**Comment by JeremiahTee** [comment]
image
image

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+branches:
+  only:
+    - develop
+    - ci-integration-44
+
+script:
+    - cd Server && npm install && npm test
+    - cd ../Client && npm install
+    - cd ../Server
```
Comment (GD-Dheer) [inline .travis.yml]: Not necessary to go back into server to do a compose

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+node_js:
+  - "14"
+
+services:
+   - docker
+
+branches:
+  only:
+    - develop
+    - ci-integration-44
```
Comment (GD-Dheer) [inline .travis.yml]: After some thought - we should run travis on develop & master

**Comment by GD-Dheer** [comment]
Screen Shot 2020-10-18 at 3 32 21 PM

Could you take a look? I triggered a new build on dev. @JeremiahTee

File: .travis.yml (hunk 1 above)
Comment (JeremiahTee) [reply .travis.yml]: Sweet! I didn't know. I removed Line 19 
Solved with commit ea46e5d

File: .travis.yml (hunk 2 above)
Comment (JeremiahTee) [reply .travis.yml]: Added master branch
Solved with commit ea46e5d

**Review by Mehrdadmaskull — APPROVED** [review]
LGTM

**Comment by JeremiahTee** [comment]
Looks like the failing build on develop is because Travis CI isn't properly set up there. As soon as this PR is approved, and ci-integration-44 branch is merged into develop, it will fix develop branch build and make it pass.

**Review by GD-Dheer — APPROVED** [review]

**Pull request merged**